### ISSUE TITLE
Segfault after v0.7 update

### ISSUE THREAD

**Issue by epassaro** [pyne/pyne#1244]
Hi!

We've updated `pyne` to v0.7 in [TARDIS-SN](https://github.com/tardis-sn/tardis) through `conda-forge` channel and our [Linux testing pipeline](https://dev.azure.com/tardis-sn/TARDIS/_build/results?buildId=2116&view=logs&j=ad07b744-5a1a-5d54-7693-fcb02e666633&t=22166bb3-3c9c-59db-871e-d8f447c89a80&l=162) throws a segmentation fault (the macOS pipeline does not fail).

As far I can see, this happens after writing HDF files.

**Comment by gonuke** [comment]
Can you point us to which TARDIS-SN test is failing so we can see how it is using PyNE?

**Comment by gonuke** [comment]
I know it's probably in the pipeline output you linked to, but we might find it faster if you can point us to it more directly.

**Comment by epassaro** [comment]
The link points directly to line 162:

```
Fatal Python error: Segmentation fault

Current thread 0x00007f76129bf700 (most recent call first):
Test results remaining: 403. Test run id: 2942
```

This happens straight after `pytest` finishes and outputs the results.

I managed to localize the problem to the `tardis/plasma/tests/test_hdf_plasma.py` test module by running them individually, so my suspicion is the segfault comes after writing HDF files.


Thank you and ping me if I can help.

**Comment by gonuke** [comment]
Thanks @epassaro - I saw the line 162, but that didn't help me find that it was in `test_hdf_plasma.py`.  I'll take a look now.

**Comment by epassaro** [comment]
No problem! It fails silently (in fact all the tests pass) maybe the segfault is thrown after exiting python?

**Comment by gonuke** [comment]
I'm afraid I don't know much about TARDIS, but I don't see anywhere that this test uses PyNE at all.  It might be deep in the dependency tree somewhere, but it appears that throughout TARDIS you only use `data`, `nucname` and `materials`.  Only `materials` does any HDF writing and I don't see that being used in this test.

So I wonder if one of our conda-forge dependencies is forcing a version change for something else that you are using and that conflict is the issue?  

(BTW, some conda-forge maintainers merged an incorrect branch before were ready and we're still trying to untangle that.)

Perhaps a quick conversation could help us get to the bottom of this?

**Comment by epassaro** [comment]
Yes sure, tomorrow or friday would be ok for me.

**Comment by gonuke** [comment]
Which time zone are you in?

**Comment by epassaro** [comment]
I'm in UTC-03, Buenos Aires.

**Comment by wkerzendorf** [comment]
I'd be happy to join on Friday at UTC+1 in Munich.

**Comment by gonuke** [comment]
OK - feel free to email me your contact info.  My email address is in my Github profile

**Comment by epassaro** [comment]
With @Rodot-'s help debugged our code using GDB, and the result was: 

```
Thread 1 "python" received signal SIGSEGV, Segmentation fault.
0x00007fffdc7f43a0 in closure_dealloc (clo=0x7fffdb74b2f0) at numba/_dynfunc.c:175
175     numba/_dynfunc.c: No such file or directory.
```

This is a known bug of `numba` 0.43 on Python >=3.7.4 (see: https://github.com/numba/numba/issues/4323) fixed in 0.45.0.

To be clear:

- Recent `pyne` update in `conda-forge` triggered Python update from 3.6 to 3.7.8.
- `numba` is aggressively pinned to 0.43 in TARDIS environment file.
- `numba` 0.43 has a bug in Python >=3.7.4 that leads to a segmentation fault.
- **This bug has nothing to do with `pyne`.**
- Upgrading `numba` to 0.45 solves the issue.

Thank you for your time @gonuke.


Closing.
